### PR TITLE
Improvement: Implement accessibility statement and feedback mechanism, resolves #567.

### LINKED ISSUE #567: accessibility statement and feedback mechanism
As already diskussed we suggest following enhancement(s) (feature) for Boost Union to improve accessibility resp. provide important accessibilty functionalities:

# accessibility statement
To be able to comply with EU Directive 2016/2102 
https://eur-lex.europa.eu/legal-content/DE/TXT/PDF/?uri=CELEX:32016L2102 
there should be the possibility for universities to offer an **accessibility statement** within Moodle, respectively with the BoostUnion theme. 
In addition to the customizable description, the accessibility statement should contain a link to electronic contact, via which users can report any deficiencies in Moodles course pages, in terms of compliance with accessibility requirements, to the relevant universities office (which can be defined by the Moodle admins in the settings) [...]." ($9b NBGG)

### Minimum requirements:
The accessibility statement and feedback mechanism are easy to find (question mark, footer button, footer)
The accessibility statement describes and directs to the feedback mechanism.
The accessibility statement and the feedback mechanism itself must be accessible. 

### sample:
grafik

# accessibility reporting button
It should be possible for admins to activate a further footerbutton, "**accessibility reporting button**" and/or footer link to feedback mechanism.

### Minimum requirements of the feedback mechanism:
The feedback mechanism (button) should be placed directly above the Moodle question mark (footer button) and should always be accessible. There should be the option to decide whether an aditional feedback link should be placed within the footer.

### sample:
grafik

### Mechanism provides automated text:
- Technical data such as operating system, browser used, etc. are pre-filled.
- The URL of the page with the barrier is transmitted automatically
- Form to answer Question if User works with assistive technologies such as a screen reader, zoom text,
   voice control or similar? If yes, please specify which ones.

# single feedback page
All contact options should be available on that single feedback page:
- Call accessibility support
- Mail accessibility support
- Form to contact accessibility support

### sample:
grafik

# 

### admin settings static page sample:
grafik
# 
### Many thanks in advance for boostunion teams efford to further improve accessibility :-)

### PR DESCRIPTION
**Pull request by detomon**
Resolves #567.

### CODE REVIEW

**Comment by christianwolters** [comment]
Hi @detomon,

I'd like to review your pull request but there are unrelated commits in this PR.
(maybe a rebase went wrong?)

Can you please cleanup your PR by resetting your branch to current main and cherry-pick your main commit?

Cheers 
Christian

**Comment by detomon** [comment]
Hi @christianwolters,

It seems that `main` has been rebase at some point. Should be ok now.

Thank you for taking the time to review our pull request!

File: lang/en/theme_boost_union.php
Comment (christianwolters) [inline lang/en/theme_boost_union.php:871]: Two different spellings - using "D" and "d" - are used on the settings page:
* here "The declaration of accessibility..."
* and later (e.g. L 874) "... shown on a Declaration of ..."

We should opt for one consistent stylization.

File: pages/accessibilitysupport.php
Comment (christianwolters) [inline pages/accessibilitysupport.php:90]: If you know the feature, and the page this is straight forward and good,
but when you are coming here later - e.g. for bug research - it would be nice to have some more comments for guidance:

- To clarify intent during the form data processing, please add some comments to the if/else branches.

File: layout/columns2.php
Comment (christianwolters) [inline layout/columns2.php:128]: Please add a remark of this modification to the phpdoc at the top of this file.

File: layout/drawers.php
Comment (christianwolters) [inline layout/drawers.php:193]: Please add a remark of this modification to the phpdoc at the top of this file.

**Review by christianwolters — CHANGES_REQUESTED** [review]
Hi @detomon,

thank you and your coworkers for this contribution. I really like seeing a11y getting a united boost here (sorry ^^)

My manuel user facing tests checked out, during code review I found a few spots that still need some work:

- I commented directly in code as you should see above

Furthermore, please take a look at your **version.php**
- you are introducing new settings, so please increment the value of $plugin->version

Cheers
Christian

**Comment by christianwolters** [comment]
## Review

Following https://github.com/moodle-an-hochschulen/moodle-plugin-maintaining/wiki/Check-list-for-peer-reviewing-patches-and-pull-requests

### Documentation

- [X] **Commit Message** understandable and issue referenced (via resolves keyword)
- [X] **Patch author** correct
- [X] **CHANGES.md** appended
- [X] **README.md** appended
- [X] **Credits** appended
- [ ] Appropriate **Comment quantity**
- [X] Successful **Moodle PHPDoc check**

### version.php

- [ ] Checked if **$plugin->version** increment necessary (and increment done if necessary)
- [X] **$plugin->release** untouched

### lib.php

- [X] Only **necessary functions**

### Languages

- [X] Only **english strings**
- [X] Necessary **magic strings** added (e.g. capabilities)

### Automated tests

- [X] **New functionality covered**
- [X] **No failing steps or scenarios**

### Mustache templates

- [X] **Example context exists**

### CSS and styles.css

- [X] **Bootstrap styles** used if possible

### Duplicated code

- [X] **Duplicated code is marked properly**

### Github action

- [X] **All green**

### Commit history and scope

- [X] Already **single commit**
- [X] Focus on **single purpose**
- [X] No **surplus files**

### Additional aspects for Boost Union

- [X] No usage of $theme->settings
- [X] Usage of isset() checks when processing plugin settings in renderer / output code
- [X] Usage of admin_setting_configselect instead of admin_setting_configcheckbox
- [X] Modified mustache templates properly marked and .upstream template in place

## Review result

Hi @detomon, here the results of my review. (I forgot to append the checklist)

see comments and conversation above, open checks:
- [ ] Appropriate **Comment quantity**
- [ ] Checked if **$plugin->version** increment necessary (and increment done if necessary)

Do you have time in the next weeks to address the open points?

Cheers
Christian

**Comment by detomon** [comment]
Hoi @christianwolters,

I just pushed the changes regarding the missing points.

Cheers,
Simon

**Comment by christianwolters** [comment]
Hi @detomon,

thank you for your changes, this looks good.

As this is PR involces changes in many files, I'll now hand over to @abias for a second review and integration.

Cheers
Christian

**Review by christianwolters — APPROVED** [review]

**Comment by abias** [comment]
Hi @detomon ,

@christianwolters handed over this PR for final review to me.
Many thanks from my side as well to you for contributing this very important feature!

I think your implementation was generally fine and production-ready, especially as I know that you are already running the code locally.
However, during my final review, I made several changes to your code and pushed them in a "Review changes" commit to your branch.

These changes were _not_ triggered because you did something wrong, it was mainly polishing and aligning the code to better fit into the existing codebase.
In addition to that, I dared to make some conceptual changes to the implementation. We could only have avoided that only if we both would have synchronized during your implementation phase.

The details of the changes are as follows, please appologize that the changes are not separated in individual commmits:
* Housekeeping (Align strings, comments and so on)
* Separate the code and the URLs for the accessbility pages from the static pages (to avoid any interference which might happen in the future)
* Give accessibility a dedicated settings page (to expose its value within the Boost Union settings and to allow more accessibility features to be added in the future)
* Use more hide_if in the accessibility settings
* Implement a setting for changing the screenreader link title
* Remove the accessibilityname setting as it is not really needed IMHO
* Split the technical information string to make the translator's life easier
* Change the way how the message subject is composed
* Replace the accessibility-support-email-notavailable template (which was a copy of core_user/contact_site_support_not_available) with a simple bootstrap notification
* Do not show the accessibility support button on the login page unless the support page is enabled
* Make sure that the accessibility support button does not overlap the communications button (which may appear in a course where communication is enabled)
* Align and enhance Behat tests

This PR will go into integration testing by the Boost Union team now before I will finally merge it.

The only thing which I would like to ask you is:
The support form comes with a support for Recaptcha as the form is a copy of the Moodle core support form. However, I am torn if adding a recaptcha really makes sense for an accessibility support form.
I would rather like to see the recaptcha support removed in the support form altogether or at least to add an admin setting to allow the admin to disable it in the accessibility support form independently from the rest of Moodle.
What would you, as the original implementer, think of this aspect?

Many thanks again,
Alex

**Review by abias — DISMISSED** [review]

**Comment by detomon** [comment]
Hi @abias,

Cool, thanks! We will review your changes to see, how we can improve for the next feature :)

The recaptcha would somewhat defy the purpose of the form to be easily accessible. At least for the logged in state. When the page is enabled without login (`theme_boost_union/enableaccessibilitysupportwithoutlogin`), then it might make sense to prevent spam.
I think making it configurable for this form only, would be a good solution. Maybe it can be a select with the options: "show no recaptcha", "show recaptcha if not logged in", "always show recaptcha". Although, the last option is debatable, maybe someone has a use case for it.

Cheers
Simon

**Comment by abias** [comment]
Hi @detomon ,

thank you very much for your feedback.
I went ahead and have implemented a setting to show / hide the re-captcha in the form now. 
The PR branch is updated and will then be merged as soon as the tests are green.

Cheers,
Alex

**Review by abias — APPROVED** [review]

**Pull request merged**
